### PR TITLE
Main menu

### PR DESCRIPTION
**Pull request by PBWiseman**
## Describe your changes

This branch will add a main menu with a play, settings and quit button. It will also have a settings and level select screen.

## Issue number and link

#8 #9 

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by PBWiseman** [comment]
Main menu code is ready for review. As mentioned on other branches the design will be handled by the design team, so this branch is just for functionality. Branch has a main menu, level selection, and settings menu. I also added the three initial levels so I could test loading them. There is also a fps display in the main menu and the Australia scene as well as a script to make it work, all three of which are named DELETE, which are just there to show the fps settings changing and will be deleted. It has default settings and also will save the users display settings across playthroughs.

**Comment by PBWiseman** [comment]
Fixed issue with aspect ratios not setting correctly

**Review by ChaseBENNC9 — APPROVED** [review]
# Code Review

## Functional Requirements
_Is the application fit for purpose? Does it meet the user’s requirements and properly address the issue?_  
Yes the issue has been properly addressed. 
 - The menu functions correctly, 
 - all the resolutions now display correctly and the FPS selection is working,
 - the level selection screen is easy to navigate and is functioning as expected 
 


## Legibility and Style
_Uses agreed coding standards/conventions? Readable, consistent and well-structured with useful comments?_  

 - Great use of comments, all code is written well and formatted for readability.

_Any repetitive code that could be refactored? (DRY)_  

 - No all code is dry


## Correctness of the Code
_Is the code reliable, giving same results every time it runs? Does it function correctly?_  

 - Yes, the code is fully functional in a build of the game with the resolution issue being fully fixed 


# Final Comments

 - Yes everything seems to be working as expected, im happy for this to be merged

***

File: Assets/Scripts/Menus/DisplayManager.cs [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+
+    /// <summary>
+    /// Loads the settings from PlayerPrefs
+    /// </summary> 
+    private void LoadSettings()
+    {
+        //To test default values uncomment the line below
+        //PlayerPrefs.DeleteAll();
+
+        //These are all inside try catch blocks as the PlayerPrefs values could have index out of range errors if the player changes monitors to one that doesn't support the resolution they had set
```
Comment (erokasyuart) [inline Assets/Scripts/Menus/DisplayManager.cs:206]: I was about to add that the resolution settings save, but none of the other settings save.

**Review by erokasyuart — CHANGES_REQUESTED** [review]
1. It would be good to have the resolution text to update when the resolution has been updated.
image

Otherwise, I like that it updates automatically instead of having an "apply" button--makes the user click less.

2. I think a back button would be useful in the character select menu in case the user changes their mind.
3. I also think that the play button should appear **after** the user has selected a continent, as the button has no functionality before. It also helps to guide the player to where to click next.

**Comment by PBWiseman** [comment]
> 1. It would be good to have the resolution text to update when the resolution has been updated.

This is just the debug text to prove the fps is changing. Have deleted this script and display.

> 2. I think a back button would be useful in the character select menu in case the user changes their mind.

Good idea. I am going to quickly change how the char select screen works to make this easier and reduce loading scenes needed.

> 3. I also think that the play button should appear **after** the user has selected a continent, as the button has no functionality before. It also helps to guide the player to where to click next.

The button is now removed until level selected.

**Comment by erokasyuart** [comment]
I did like the text showing what the settings were... kinda like the version text at the bottom corner of the menu on other games, but I guess it is a bit debuggy

**Review by erokasyuart — APPROVED** [review]

**Comment by PBWiseman** [comment]
Display settings now save and the character select scene has been merged with the main menu one so I can implement a back button without constantly changing scenes.

File: Assets/Scripts/Menus/DisplayManager.cs (hunk 1 above)
Comment (PBWiseman) [reply Assets/Scripts/Menus/DisplayManager.cs:202]: Resolved

**Review by ChaseBENNC9 — APPROVED** [review]
Resolutions now save when quitting  Character select has back button, changes look good

**Review by erokasyuart — CHANGES_REQUESTED** [review]
When you unselect a continent, the play button remains, and the player can enter character with a place unselected.

**Review by erokasyuart — APPROVED** [review]

**Pull request merged**